lib/ts/ink_inet.h

```diff
@@ -620,7 +620,7 @@ ats_is_ip_loopback(IpEndpoint const *ip)
 inline bool
 ats_is_ip_multicast(sockaddr const *ip)
 {
-  return ip && ((AF_INET == ip->sa_family && 0xe == *ats_ip_addr8_cast(ip)) ||
+  return ip && ((AF_INET == ip->sa_family && 0xe == (*ats_ip_addr8_cast(ip) >> 4)) ||
                 (AF_INET6 == ip->sa_family && IN6_IS_ADDR_MULTICAST(&ats_ip6_addr_cast(ip))));
 }
```

Comment:
See [here](https://github.com/apache/trafficserver/blob/3fc04c4590d7e47ba648b06f973c874d6ff900fc/lib/ts/ink_inet.cc#L465) where we similarly shift 4 bits.

This is intended to compare `224.0.0.0/4`, the high 4 bits of an IPv4 [multicast address](https://en.wikipedia.org/wiki/Multicast_address#IPv4).

Reply:
`0xE0 == *ats_ip_addr8_cast(ip) & 0xF0`?

Reply:
Sure; I did the shift because [ink_inet.cc](https://github.com/apache/trafficserver/blob/3fc04c4590d7e47ba648b06f973c874d6ff900fc/lib/ts/ink_inet.cc#L465) did it that way, and I have no problem changing this PR if we'd rather mask.

Reply:
Leaving it be, as we discussed in person.

Reply:
I would do `ats_ip_addr8_cast(ip)[0])` instead of `*ats_ip_addr8_cast(ip)` to show it is an array.  It is done in another place in the code and it should be consistent.

Reply:
Agree. Will change it.

Reply:
Addressed.